numba_cuda/numba/cuda/cuda_bf16.py

```diff
@@ -0,0 +1,5070 @@
+# Automatically generated by Numbast Static Binding Generator
+# Generator Information:
+# Ast_canopy version: 0.3.0
+# Numbast version: 0.3.0
+# Generation command: /home/wangm/numbast/numbast/src/numbast/__main__.py --cfg-path configs/cuda_bf16.yml --output-dir numba_cuda/numba/cuda/
+# Static binding generator parameters: {'cfg_path': 'configs/cuda_bf16.yml', 'output_dir': 'numba_cuda/numba/cuda/', 'input_header': None, 'retain': None, 'types': None, 'datamodels': None, 'compute_capability': None, 'run_ruff_format': True}
+# Config file path (relative to the path of the generated binding): ../../../../configs/cuda_bf16.yml
+# Cudatoolkit version: (12, 8)
+# Default CUDA_HOME path: /home/wangm/micromamba/envs/numbast
+
+
+# Imports:
+from numba.types import float64
+from numba.types import int32
+from numba.cuda.cudadecl import register
+from numba.types import uint8
+from numba import types
+from numba.types import uint64
+from numba.cuda import declare_device
+from numba.types import float16
+import operator
+from numba.core.datamodel.old_models import PrimitiveModel
+from numba.core.typing.templates import ConcreteTemplate
+from numba.types import CPointer
+from numba.types import int64
+from numba.types import bool_
+from numba.types import int8
+from numba.core.typing import signature
+from numba.core.extending import register_model
+from numba.types import int16
+import numba
+from numba.core.extending import lower_cast
+from numba.extending import as_numba_type
+from numba.types import Number
+from llvmlite import ir
+from numba.cuda.cudadecl import register_global
+from numba.types import Type
+from numba.types import uint16
+from numba.core.datamodel.old_models import StructModel
+from numba.cuda.cudadecl import register_attr
+from numba.cuda import CUSource
+from numba.core.extending import make_attribute_wrapper
+import io
+from numba.core.typing.templates import AttributeTemplate
+from numba.cuda.cudaimpl import lower
+from numba.types import float32
+from numba.types import uint32
+from numba.types import Function
+
+
+# Prefixes:
+
+numba.config.CUDA_ENABLE_PYNVJITLINK = True
+
+# Shim Stream:
+
+entry_point = "#include </usr/local/cuda/include/cuda_bf16.h>"
```

Comment:
We discussed offline that it would be more reliable if we vend the cuda_bf16 header in Numba and generate binding for it. This means:
1. If there's a new CTK major release and Numba supports it, the generated binding whose shims contains implementation from older CTK release is still supported. 
2. When we want to also upgrade the bindings, we would need to re-vend the header and re-generate the bindings as well.